docs/overview/data-flow.md

```diff
@@ -0,0 +1,92 @@
+# Data Flow
+
+On this page, we describe the data flow inside Vald and how vector indexes are stored in the Vald Cluster.
+It will help you to understand what Vald does from users' requests.
+
+The below image is the basic Vald architecture.
+
+<img src="../../assets/image/svg/Vald Architecture Overview.svg" />
+
+We will explain using this image in the following sections.
+
+## Insert
+
+Users can insert a vector to Vald cluster using Insert API. When the insert command is processed, Vald will intelligently insert the index into the most suitable Vald Agent(s) base on the memory usage of the Vald Agent and the node.
+
+To obtain the memory usage of Vald Agent and the node, Vald Discoverer is required to rank which is the most suitable agent to perform the insert request by talking with the kube-apiserver.
+
+To make the insert command effective, the `CreateIndex` instruction is required by the Vald Index Manager or Vald Agent itself to update the index to provide extremely high searching performance to users.
+
+Please note that only one embedding space is supported in a single Vald Cluster, if you want to support multiple embedded spaces, you may need to consider to deploy multiple Vald cluster to support this use case.
```

Comment:
We should mention about the current Vald can handle only a single embedding space.

Reply:
I think I have already mentioned here 🤔
`Please note that only one embedding space is supported in a single Vald Cluster`